src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -125,6 +142,7 @@ static void mlpackMain()
   {
     ReportIgnoredParam({{ "input_labels", true }}, "training_labels");
     ReportIgnoredParam({{ "input_labels", true }}, "test_labels");
+    ReportIgnoredParam("stratify_data", "input_labels is not provided");
```

Comment:
I don't think that this works as intended.  Can you check the documentation for `ReportIgnoredParam`?

Reply:
What I wanted to do here was to provide the user a warning if he provides the ```stratify_data``` parameter without providing the ```input_labels``` file. 
In this case, the ```stratify_data``` parameter is ignored.
Since no labels are passed to stratify from, we go ahead and perform a normal split. 

I went through the ```mlpack::util``` docs and the ```ReportIgnoredParam``` seemed like the function to use.
According to the ```ReportIgnoredParam``` docs, 
```If the given parameter is passed, report that it is ignored, supplying a custom reason.```

In this case, the ```stratify_data``` parameter is ignored because the ```input_labels``` parameter is not provided. 

Since this call is made inside the ```else``` block where the ```if``` checks if the ```input_labels``` file is provided, I thought this would be the right place to include it.

Let me know if this was not the right way to do it.